src/routers/xcm/Router.sol

```diff
@@ -25,13 +26,63 @@ contract ConnectorXCMRouter {
     using TypedMemView for bytes29;
     using ConnectorMessages for bytes29;
 
+    /// --- Properties ---
     ConnectorLike public immutable connector;
-
     address centrifugeChainOrigin;
```

Comment:
We should make `centrifugeChainOrigin` and `centrifugeChainHandleCallIndex` immutable. This makes reading this data cheaper, and if we ever need to update these we could swap the entire router instead.

Reply:
So instead of reading the data from the EVM we just hardcode it? The issue with that is that - at least the call index - will be a different value on the Centrifuge runtime vs the development runtime we are now testing against. Let me know 👍

Reply:
We don't need to hardcode it. Just add the `immutable` modifier to the storage value, like the `connector` storage variable. Then, the EVM is smart enough to only allow updating the value in the constructor, but never afterwards and then treating it like a constant.

Reply:
Some context: https://ethereum.stackexchange.com/questions/82240/what-is-the-immutable-keyword-in-solidity

Reply:
I see. I had tried to make the call index but that doesn't seem possible because of its `bytes` type:

```
error[6377]: src/routers/xcm/Router.sol:32:5: TypeError: Immutable variables cannot have a non-value type.
    bytes immutable centrifugeChainHandleCallIndex;
```

Reply:
Ah. Any chance we can map from a uint256 value to this bytes value? Or some other mapping from a value type?

Mapping would almost certainly still be cheaper than reading from storage.

I think it's also fine to address this in a followup PR and add an issue though.

Reply:
Actually, reading the docs, it looks like it would work if you change it to `bytes32` (or smaller if that fits)

Reply:
Done: https://github.com/centrifuge/connectors/issues/47

Reply:
I would try `bytes32` if that works. That would be a small change and could fix it. If not, we tackle it later.

Reply:
that wouldn't work, I will tackle it in a follow up PR 👍  I have an idea to tackle it nicely

Reply:
@offerijns just tackled this here now: https://github.com/centrifuge/connectors/pull/46/commits/fcf260677f54e086046732ce372416595a0a98b4

Basically we make the pallet index and the extrinsic index two seperate `uint8` params and  thus `immutable properties of the contract 👍